### ISSUE TITLE
Using externally provided third-party libraries

### ISSUE THREAD

**Issue by paulromano** [openmc-dev/openmc#1779]
Along the same lines as #1778, OpenMC should allow using third-party libraries that are not vendored in order to play nicely when it is built as part of a larger software stack. We recently started allowing this for pugixml and fmt, but xtensor, xtl, and gsl-lite are forced to be our own vendored versions.

I'll note that this is not a hypothetical issue -- we've run into problems recently with builds of ENRICO, which may include both OpenMC and Shift, having conflicting copies of pugixml and gsl-lite.

**Comment by paulromano** [comment]
One possible solution: https://github.com/cpm-cmake/CPM.cmake

Also some chatter on the cmake repo about enabling a seamless integration of `FetchContent` and `find_package` (equivalent to what the above CPM.cmake package currently provides).

**Comment by paulromano** [comment]
Just to reframe this a little bit, there's really two separate parts:
1. Add `find_package` calls for `xtensor` and `gsl-lite` akin to what was done in #1757 for pugixml and fmt.
2. Consider replacing our git submodules for dependencies with [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html) (or something fancier built on top of that like [CPM](https://github.com/cpm-cmake/CPM.cmake))

**Comment by paulromano** [comment]
If the `find_package` call succeeds, I don't think we should be using the submodule. Although it's a little weird to find the package in an existing OpenMC installation, I don't see a good way around that (hard to know where the package was picked up from).

I was able to reproduce that error you're seeing, and it looks like we should be using the namespaced name for gsl-lite-v1 in target_link_libraries:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d588cb84..38069662a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -446,7 +446,7 @@ endif()
 # target_link_libraries treats any arguments starting with - but not -l as
 # linker flags. Thus, we can pass both linker flags and libraries together.
 target_link_libraries(libopenmc ${ldflags} ${HDF5_LIBRARIES} ${HDF5_HL_LIBRARIES}
-                      pugixml faddeeva xtensor gsl-lite-v1 fmt::fmt)
+                      pugixml faddeeva xtensor gsl::gsl-lite-v1 fmt::fmt)
 
 if(dagmc)
   target_compile_definitions(libopenmc PRIVATE DAGMC)
```

**Comment by kkiesling** [comment]
Presumably, the `NO_SYSTEM_ENVIRONMENT_PATH` option in `find_package` should be skipping the install that already exists from the previous openmc build, but that does not seem to be the case when I build. I have removed the install path from my `PATH` and `LD_LIBRARY_PATH` even. And I have tried all the other `NO_*` options for `find_package` to attempt to exclude that path. I am unsure why it continues to be found. There doesn't seem to be any way to exclude a _specific_ path either (then it would be easy to just exclude `CMAKE_INSTALL_PREFIX`).

**Comment by gridley** [comment]
This solution is maybe a little dirty, but we could set up all the submodules as forks under `github.com/openmc-dev`, then just tweak the namespaces of the modules held in OpenMC's forks, for instance using `openmc::fmt` in order to avoid collisions with the system's version.

**Comment by paulromano** [comment]
@kkiesling Yeah, I found the behavior you're seeing a little strange too. To reproduce what you were seeing, I explicitly had to add my install directory to `CMAKE_PREFIX_PATH`. If you haven't already, you may want to try adding `-DCMAKE_FIND_DEBUG_MODE=ON` so that you can see where it's searching for packages (the order may be indicative of why it's picking up the installed version).

@gridley I understand the motivation to tweak things for GPU porting purposes, but I'd rather not be in the game of managing our own versions of these dependencies. Each time we want to update our own version, there will be potential conflicts to be resolved.

**Comment by kkiesling** [comment]
my cmake line included `-DCMAKE_INSTALL_PREFIX` (not `CMAKE_PREFIX_PATH`, which doesn't appear to be in my `CMakeCache.txt` even).

**Comment by paulromano** [comment]
Closed by #2006
